### PR TITLE
vic-machine suggests valid datastore

### PR DESCRIPTION
**Pull request by andrewtchin**
ESX
```
⇒  bin/vic-machine-linux create --target 192.168.218.177 --name chin-vic-110 --user root --password password --image-datastore datastore/asdf --compute-resource asdf --volume-store /ha-datacenter/datastore/datastore1/volpath1:vollabel1 --volume-store datastore/volpath2:vollabel2
INFO[2016-08-04T10:04:49-07:00] ### Installing VCH ####                      
INFO[2016-08-04T10:04:49-07:00] Generating certificate/key pair - private key in ./chin-vic-110-key.pem 
INFO[2016-08-04T10:04:49-07:00] Validating supplied configuration            
INFO[2016-08-04T10:04:49-07:00] Suggesting valid values for --compute-resource based on "/ha-datacenter/host/*/Resources/asdf" 
INFO[2016-08-04T10:04:49-07:00] Suggested values for --compute-resource:          
INFO[2016-08-04T10:04:49-07:00]   "localhost.localdomain/chin-vic-32"        
INFO[2016-08-04T10:04:49-07:00]   "localhost.localdomain/docker-appliance"   
INFO[2016-08-04T10:04:49-07:00] Suggesting valid values for --image-datastore based on "datastore/asdf" 
INFO[2016-08-04T10:04:49-07:00] Suggested values for --image-datastore:      
INFO[2016-08-04T10:04:49-07:00]   "datastore1"                               
INFO[2016-08-04T10:04:49-07:00] Suggesting valid values for --volume-store based on "*:vollabel1" 
INFO[2016-08-04T10:04:49-07:00] Suggested values for --volume-store:         
INFO[2016-08-04T10:04:49-07:00]   "datastore1:vollabel1"                     
INFO[2016-08-04T10:04:49-07:00] Suggesting valid values for --volume-store based on "datastore/volpath2:vollabel2" 
INFO[2016-08-04T10:04:50-07:00] Suggested values for --volume-store:         
INFO[2016-08-04T10:04:50-07:00]   "datastore1:vollabel2"                     
INFO[2016-08-04T10:04:50-07:00] Firewall status: DISABLED on "/ha-datacenter/host/localhost.localdomain/localhost.localdomain" 
INFO[2016-08-04T10:04:50-07:00] Firewall configuration OK on hosts:          
INFO[2016-08-04T10:04:50-07:00]   "/ha-datacenter/host/localhost.localdomain/localhost.localdomain" 
WARN[2016-08-04T10:04:50-07:00] Evaluation license detected. VIC may not function if evaluation expires or insufficient license is later assigned. 
INFO[2016-08-04T10:04:50-07:00] License check OK                             
INFO[2016-08-04T10:04:50-07:00] DRS check SKIPPED - target is standalone host 
ERRO[2016-08-04T10:04:50-07:00] --------------------                         
ERRO[2016-08-04T10:04:50-07:00] resource pool '/ha-datacenter/host/*/Resources/asdf' not found 
ERRO[2016-08-04T10:04:50-07:00] datastore 'datastore' not found              
ERRO[2016-08-04T10:04:50-07:00] datastore hostname empty                     
ERRO[2016-08-04T10:04:50-07:00] datastore 'datastore' not found              
ERRO[2016-08-04T10:04:50-07:00] Create cannot continue: configuration validation failed 
ERRO[2016-08-04T10:04:50-07:00] --------------------                         
ERRO[2016-08-04T10:04:50-07:00] vic-machine-linux failed: validation of configuration failed
```

vC
```
⇒  bin/vic-machine-linux create --target 10.17.109.115 --image-datastore istore/asdf --volume-store vol1:label1 --volume-store vol2/asdf:label2 --name chin-vic-18 --user administrator@vsphere.local --password password --compute-resource cluster1/asdf --bridge-network vch-test --timeout 10m
INFO[2016-08-04T10:06:23-07:00] ### Installing VCH ####                      
INFO[2016-08-04T10:06:23-07:00] Generating certificate/key pair - private key in ./chin-vic-18-key.pem 
INFO[2016-08-04T10:06:28-07:00] Validating supplied configuration            
INFO[2016-08-04T10:06:30-07:00] Suggesting valid values for --compute-resource based on "/dc1/host/cluster1/Resources/asdf" 
INFO[2016-08-04T10:06:30-07:00] Suggested values for --compute-resource:     
INFO[2016-08-04T10:06:30-07:00]   "cluster1/chin"                            
INFO[2016-08-04T10:06:31-07:00] Suggesting valid values for --image-datastore based on "istore/asdf" 
INFO[2016-08-04T10:06:31-07:00] Suggested values for --image-datastore:      
INFO[2016-08-04T10:06:31-07:00]   "vsanDatastore"                            
INFO[2016-08-04T10:06:31-07:00] Suggesting valid values for --volume-store based on "vol1:label1" 
INFO[2016-08-04T10:06:32-07:00] Suggested values for --volume-store:         
INFO[2016-08-04T10:06:32-07:00]   "vsanDatastore:label1"                     
INFO[2016-08-04T10:06:32-07:00] Suggesting valid values for --volume-store based on "vol2/asdf:label2" 
INFO[2016-08-04T10:06:32-07:00] Suggested values for --volume-store:         
INFO[2016-08-04T10:06:32-07:00]   "vsanDatastore:label2"                     
INFO[2016-08-04T10:06:34-07:00] vDS configuration OK on "vch-test"           
INFO[2016-08-04T10:06:36-07:00] Firewall status: ENABLED on "/dc1/host/cluster1/10.17.109.74" 
INFO[2016-08-04T10:06:37-07:00] Firewall status: ENABLED on "/dc1/host/cluster1/10.17.109.76" 
INFO[2016-08-04T10:06:39-07:00] Firewall status: ENABLED on "/dc1/host/cluster1/10.17.109.169" 
INFO[2016-08-04T10:06:39-07:00] Firewall configuration OK on hosts:          
INFO[2016-08-04T10:06:39-07:00]   "/dc1/host/cluster1/10.17.109.74"          
INFO[2016-08-04T10:06:39-07:00]   "/dc1/host/cluster1/10.17.109.76"          
INFO[2016-08-04T10:06:39-07:00]   "/dc1/host/cluster1/10.17.109.169"         
INFO[2016-08-04T10:06:41-07:00] License check OK on hosts:                   
INFO[2016-08-04T10:06:41-07:00]   "/dc1/host/cluster1/10.17.109.74"          
INFO[2016-08-04T10:06:41-07:00]   "/dc1/host/cluster1/10.17.109.76"          
INFO[2016-08-04T10:06:41-07:00]   "/dc1/host/cluster1/10.17.109.169"         
INFO[2016-08-04T10:06:41-07:00] DRS check OK on:                             
INFO[2016-08-04T10:06:41-07:00]   "/dc1/host/cluster1/Resources"             
ERRO[2016-08-04T10:06:42-07:00] --------------------                         
ERRO[2016-08-04T10:06:42-07:00] resource pool '/dc1/host/cluster1/Resources/asdf' not found 
ERRO[2016-08-04T10:06:42-07:00] datastore 'istore' not found                 
ERRO[2016-08-04T10:06:42-07:00] datastore 'vol1' not found                   
ERRO[2016-08-04T10:06:42-07:00] datastore 'vol2' not found                   
ERRO[2016-08-04T10:06:42-07:00] Create cannot continue: configuration validation failed 
ERRO[2016-08-04T10:06:42-07:00] --------------------                         
ERRO[2016-08-04T10:06:42-07:00] vic-machine-linux failed: validation of configuration failed
```

### CODE REVIEW

File: lib/install/validate/storage.go [hunk 1]
```diff
@@ -31,39 +32,34 @@ func (v *Validator) storage(ctx context.Context, input *data.Data, conf *config.
 	defer trace.End(trace.Begin(""))
 
 	// Image Store
-	imageDSpath, ds, err := v.DatastoreHelper(ctx, input.ImageDatastoreName)
+	imageDSpath, ds, err := v.DatastoreHelper(ctx, input.ImageDatastoreName, "", "--image-datastore")
 	v.NoteIssue(err)
-	if ds == nil {
-		log.Errorf("Datastore for images is nil")
-		return
+	if ds != nil {
```
Comment (fdawg4l) [inline lib/install/validate/storage.go]: This idiom looks a little strange.  Why not just check for error and bale out?

Comment (andrewtchin) [inline lib/install/validate/storage.go]: In vic-machine we prefer to find all issues with the user's input in one pass so they don't have to fix one thing at a time then rerun the command. Here I wanted to check volume stores also even if the image store is incorrect. Can anyone suggest if there is a better way?

**Comment by matthewavery** [comment]
lgtm

**Comment by andrewtchin** [comment]
♻️

**Pull request merged**